camayoc/qcs_models.py

```diff
@@ -437,11 +475,17 @@ def equivalent(self, other):
                 'Scan objects or dictionaries.'
             )
 
-        for key, value in self.fields().items():
-            if key == 'source':
-                if value != other.get(key).get('id'):
+        local_items = self.fields()
+        local_keys = local_items.keys()
+        other_keys = other.keys()
+        all_keys = set().union(*[local_keys, other_keys])
+        for key in all_keys:
+            if key == 'status':
+                continue
+            if key == 'sources':
+                if sorted(local_items.get(key)) != sorted(other.get(key)):
                     return False
-            else:
-                if not other.get(key) == value:
+            if key not in ['status', 'sources']:
+                if not other.get(key) == local_items.get(key):
```

Comment:
since key should be defined, you could use other[key] and local_items[key], easy to type :smile: